lib/connector/OAuthConnector.js

```diff
@@ -272,19 +272,29 @@ class OAuthConnector {
    * Deletes user context from storage.
    * @param {FusebitContext} fusebitContext The Fusebit context
    * @param {string} vendorUserId The vendor user id
-   * @param {string} foreignVendorId If specified, vendorUserId represents the identity of the user in another system.
-   * The foreignVendorId must correspond to an entry in userContext.foreignOAuthIdentities.
+   * @param {string} vendorId If specified, vendorUserId represents the identity of the user in another system.
+   * The vendorId must correspond to an entry in userContext.foreignOAuthIdentities.
    */
-  async deleteUser(fusebitContext, vendorUserId, foreignVendorId) {
-    const userContext = await this.getUser(fusebitContext, vendorUserId, foreignVendorId);
+  async deleteUser(fusebitContext, vendorUserId, vendorId) {
+    const userContext = await this.getUser(fusebitContext, vendorUserId, vendorId);
     if (userContext && userContext.foreignOAuthIdentities) {
       for (let fvId in userContext.foreignOAuthIdentities) {
         await fusebitContext.storage.delete(
           this.getStorageIdForVendorUser(userContext.foreignOAuthIdentities[fvId].userId, fvId)
         );
       }
     }
-    return fusebitContext.storage.delete(this.getStorageIdForVendorUser(userContext.vendorUserId));
+    return userContext && fusebitContext.storage.delete(this.getStorageIdForVendorUser(userContext.vendorUserId));
```

Comment:
At the top of the function after getUser add an `if (!userContext) { return; }` to remove a clause from both of these statements.